### PR TITLE
54818 Columns spacing updates

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR updates the appointment list column spacing as per spec.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#54818

## Testing done
Unit testing

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

| | Before | After |
| --- | --- | --- |
| Mobile | | |
| Desktop | | |

## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by simiadebowale — COMMENTED** [review]
some of the dimensions are a little off from the design.
(Web)
Max width of Type of care column in design: 285px
Max width of Type of care column in PR: 342.92px
(Mobile)
height of appt card in design: 124px
height of appt card in PR: 104px

File: src/applications/vaos/appointment-list/sass/styles.scss [hunk 1]
```diff
@@ -153,42 +156,89 @@ li.vaos-appts__listItem--clickable[data-status="pending"]:last-child {
...
+  }
+}
+
+.vaos-appts__display--table {
+  @include media($small-screen) {
+    display: table;
+    table-layout: fixed;
+    width: 100%;
+    max-width: initial;
   }
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/sass/styles.scss:224]: Removing the `@include media` but keeping its content will apply the styling to all screen sizes. 
```suggestion
.vaos-appts__display--table {
    display: table;
    table-layout: fixed;
    width: 100%;
    max-width: initial;
```

**Review by jenniemc — COMMENTED** [review]
It's coming along very well. I just have a few suggestions. Adding a screenshot for context. 

Screenshot 2023-04-06 at 7 28 53 PM

File: src/applications/vaos/appointment-list/sass/styles.scss [hunk 2]
```diff
@@ -153,42 +156,89 @@ li.vaos-appts__listItem--clickable[data-status="pending"]:last-child {
...
-  @media (min-width: $small-screen) {
-    // margin-left: 95px !important;
+.vaos-appts__listItem--lineHeight {
+  @include media($small-desktop-screen) {
+    line-height: 3.5;
   }
 }
+.medium-screen\:vaos-appts__minWidth,
+.medium-screen\:vaos-appts__maxWidth {
+  min-width: 24;
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/sass/styles.scss:191]: Done

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx [hunk 3]
```diff
@@ -95,48 +138,66 @@ export default function AppointmentColumnLayout({
...
+                'vaos-appts__column-gap--3',
+                'small-screen:vads-u-flex-direction--column',
+                'small-desktop-screen:vads-u-flex-direction--row',
+              )}
+            >
               <AppointmentColumn
                 padding="0"
                 size="1"
-                className="vads-u-font-weight--bold vaos-appts__text--truncate"
+                className="vads-u-font-weight--bold vaos-appts__display--table vaos-appts__text--truncate"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:151]: Having 'vaos-appts__text--truncate' on the column deals with text added directly to the column.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx [hunk 4]
```diff
@@ -95,48 +138,66 @@ export default function AppointmentColumnLayout({
...
+                >
+                  {appointmentLocality}
+                </span>
               </AppointmentColumn>
 
               <AppointmentColumn
                 padding="0"
                 size="1"
-                className="vaos-appts__text--truncate small-desktop-screen:vads-u-margin-left--5"
+                className="vaos-appts__display--table vaos-appts__text--truncate"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:169]: Having 'vaos-appts__text--truncate' on the column deals with text added directly to the column.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx [hunk 5]
```diff
@@ -95,48 +138,66 @@ export default function AppointmentColumnLayout({
...
+                    'vaos-appts__text--truncate',
+                  )}
+                >
+                  {appointmentLocality}
+                </span>
               </AppointmentColumn>
 
               <AppointmentColumn
                 padding="0"
                 size="1"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:168]: Done.

File: src/applications/vaos/appointment-list/sass/styles.scss (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/sass/styles.scss:224]: Done.

File: src/applications/vaos/appointment-list/sass/styles.scss [hunk 6]
```diff
@@ -153,42 +156,89 @@ li.vaos-appts__listItem--clickable[data-status="pending"]:last-child {
...
-    text-overflow: ellipsis !important;
+    max-width: 221px;
+  }
+  @include media($small-screen) {
+    max-width: initial;
   }
   @include media($medium-screen) {
-    max-width: 222px;
-    white-space: nowrap;
+    max-width: 494px;
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/sass/styles.scss:241]: Done.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:151]: fix style to apply truncation to children.

**Review by jenniemc — COMMENTED** [review]
This is fantastic! Works perfectly in the upcoming appointment list. Just need to add the truncation classes to pending appointment list.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx [hunk 7]
```diff
@@ -37,81 +37,82 @@ export default function RequestAppointmentLayout({ appointment }) {
...
+                    size="1"
+                    canceled={isCanceled}
+                    className="vads-u-font-weight--bold"
+                  >
+                    {typeOfCareName}
+                  </AppointmentColumn>
+                  <AppointmentColumn
+                    padding="0"
+                    size="1"
+                    className="vaos-appts__text--truncate"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx]: Done

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx [hunk 8]
```diff
@@ -37,81 +37,82 @@ export default function RequestAppointmentLayout({ appointment }) {
...
-                </AppointmentColumn>
-              </AppointmentRow>
-            </AppointmentColumn>
+                      {`${modality}`}
+                    </>
+                  </AppointmentColumn>
+                  <AppointmentColumn
+                    padding="0"
+                    size="1"
+                    className="vaos-appts__text--truncate"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx]: Done.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx [hunk 9]
```diff
@@ -37,81 +37,82 @@ export default function RequestAppointmentLayout({ appointment }) {
...
+            size="1"
+          >
+            <AppointmentRow className="vaos-appts__column-gap--3 small-screen:vads-u-flex-direction--row">
+              <AppointmentColumn size="1" className="vads-u-flex--4">
+                <AppointmentRow className="vaos-appts__column-gap--3 xsmall-screen:vads-u-flex-direction--column small-screen:vads-u-flex-direction--row">
+                  <AppointmentColumn
+                    padding="0"
+                    size="1"
+                    canceled={isCanceled}
+                    className="vads-u-font-weight--bold"
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx]: Done.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx [hunk 10]
```diff
@@ -37,81 +37,82 @@ export default function RequestAppointmentLayout({ appointment }) {
...
+              'vads-u-padding-y--2',
+              {
+                'vads-u-border-top--1px': grouped && !first,
+              },
+            )}
+            size="1"
+          >
+            <AppointmentRow className="vaos-appts__column-gap--3 small-screen:vads-u-flex-direction--row">
+              <AppointmentColumn size="1" className="vads-u-flex--4">
+                <AppointmentRow className="vaos-appts__column-gap--3 xsmall-screen:vads-u-flex-direction--column small-screen:vads-u-flex-direction--row">
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx]: Done.

**Review by rhasselle-oddball — APPROVED** [review]
approval for `simple-forms` file

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**